### PR TITLE
[DEV APPROVED] Add site search to FinCap

### PR DESCRIPTION
**Pull request by MarcoBgn**
[TP Card](https://moneyadviceservice.tpondemand.com/restui/board.aspx?#page=userstory/9508)

### Context

Add back-end site search capabilities, using the same features / constraints in place on the MAS site.

### Considerations

- VCR cassettes for Algolia requests included in the PR 

- The feature uses the [site_search](https://github.com/moneyadviceservice/site_search) gem 
   The Gemfile points to the feature branch at the moment, this will be changes as soon as the Gem
   is released

- Add `config/breakman.ignore` in order to add a false positive ( weak confidence warning) 
   Please refer to [this issue](https://github.com/presidentbeef/brakeman/issues/1154) for more info

### CODE REVIEW

File: app/presenters/search_result_presenter.rb [hunk 1]
```diff
@@ -0,0 +1,2 @@
+class SearchResultPresenter < BasePresenter
+end
```
Comment (tomas-stefano) [inline app/presenters/search_result_presenter.rb]: I think after the https://github.com/moneyadviceservice/site_search/pull/2 you can remove this class.

File: app/views/search_results/_search_result.html.erb [hunk 2]
```diff
@@ -0,0 +1,9 @@
+<% present(search_result, SearchResultPresenter) do |search_result| %>
```
Comment (tomas-stefano) [inline app/views/search_results/_search_result.html.erb]: I think after the https://github.com/moneyadviceservice/site_search/pull/2 you can remove the present call.

File: features/site_search.feature [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+  Scenario: Search pages with description
+    When I search the query "Financial" on top bar
+    Then I should see "7" results
+    And I should see the search results
+      | title                      |
+      | Financial well-being: the employee view |
+    And I should see the description
+       | This is an example description about a specific evidence summary. |
+
+
```
Comment (tomas-stefano) [inline features/site_search.feature]: I would remove some extra lines here.

File: features/step_definitions/search_steps.rb [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+Then('I click on the {string} button') do |type|
+  search_results_page.pagination.send("#{type}_button").click
+end
+
+Then('I should see no search results for {string}') do |query|
+  expect(search_results_page).to have_no_results
+end
+
+Then('I should see the description') do |table|
+
```
Comment (tomas-stefano) [inline features/step_definitions/search_steps.rb]: I would remove this empty line here.

File: app/assets/stylesheets/page_specific/_search_results.scss [hunk 5]
```diff
@@ -0,0 +1,4 @@
+.pagination {
+  margin: $baseline-unit*8 0 $baseline-unit*6 0;
+  align-content: center
+}
```
[the remark this answers is not in the document]
Comment (willhall88) [reply app/assets/stylesheets/page_specific/_search_results.scss]: We've already got styling of `.pagination` class in _pagination.scss. Might be worth taking this out of this PR and I can pick up the styling separately. just so this PR can be tested and merged in. I don't know whether this styling will affect other pages with pagination as well :)

File: app/views/search_results/index.html.erb [hunk 6]
```diff
@@ -0,0 +1,10 @@
+<div class="l-2col-main">
```
Comment (willhall88) [inline app/views/search_results/index.html.erb]: After our discussion yesterday, I think this file is redundant and unused? if that's correct can you remove this from the PR

**Comment by willhall88** [comment]
Two comments that need addressing, but after that it looks good :) Nice work

**Comment by MarcoBgn** [comment]
@willhall88 I have removed the .scss file, thanks for spotting it

**Pull request merged**